### PR TITLE
Fix #9711 - Investigate OracleRelationalCommandBuilderFactory 'hacks'…

### LINKED ISSUE #9711: Relational: Investigate OracleRelationalCommandBuilderFactory "hacks" in Oracle sample
Some ad hoc type mapping was required in OracleRelationalCommandBuilderFactory. We should make corresponding improvements to Relational/Core to remove the need for these changes.

### PR DESCRIPTION
**Pull request by anpete**
… in Oracle sample

- Templatize some logic in RelationalCommandBuilderFactory.

### CODE REVIEW

File: src/EFCore.Relational/Storage/Internal/RelationalCommand.cs [hunk 1]
```diff
@@ -374,5 +380,13 @@ public class RelationalCommand : IRelationalCommand
 
             return command;
         }
+
+        /// <summary>
+        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        protected virtual void ConfigureCommand(DbCommand command)
```
Comment (ajcvickers) [inline src/EFCore.Relational/Storage/Internal/RelationalCommand.cs]: Feels like this should not be documented as internal if the intention is for providers to override it. Maybe give it the text block we use elsewhere for things that we expect only providers to use?

**Comment by ajcvickers** [comment]
Also, seems like there is still code that needs cleaning up here, so suggest we still leave #9711 open.

File: src/EFCore.Relational/Storage/Internal/RelationalCommand.cs (hunk 1 above)
Comment (anpete) [reply src/EFCore.Relational/Storage/Internal/RelationalCommand.cs]: It is optional to override, as are the other methods in this class (which currently have the same comment). Want to change it for all?

Comment (ajcvickers) [reply src/EFCore.Relational/Storage/Internal/RelationalCommand.cs]: No, leave it for now. I think I need to understand better how providers are supposed to interact with this internal class.

**Pull request merged**